include/libtorrent/extensions.hpp

```diff
@@ -437,8 +437,20 @@ namespace libtorrent {
 		virtual bool on_reject(peer_request const&) { return false; }
 		virtual bool on_suggest(piece_index_t) { return false; }
 
+		virtual void sent_have_all() {}
+		virtual void sent_have_none() {}
+		virtual void sent_reject_request(piece_index_t, int, int) {}
+		virtual void sent_allow_fast(piece_index_t) {}
+		virtual void sent_suggest(piece_index_t) {}
+		virtual void sent_cancel(piece_index_t, int, int) {}
+		virtual void sent_request(piece_index_t, int, int) {}
```

Comment:
the `peer_request` type is public, so you may as well pass it as the argument for `sent_reject_request()`, `sent_cancel()` and `sent_request()`